rust/osm2lanes/src/tags.rs

```diff
@@ -105,11 +114,69 @@ impl ToString for Tags {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct TagTree(BTreeMap<String, TagTreeVal>);
```

Comment:
Some docs / examples? This would take something like `cycleway:left:separation:right` and recursively split it into pieces? So if you also had `cycleway:lift:separation:width`, you could traverse the tree and see `right` and `width` in the same place?

Reply:
correct. docs/tests do seem like a good idea :)